cmd/snapd/main.go

```diff
@@ -64,6 +90,12 @@ func run() error {
 
 	d.Start()
 
+	if ticker, err := runWatchdog(d); err != nil {
+		logger.Noticef("cannot run software watchdog: %s", err)
+	} else {
+		defer ticker.Stop()
```

Comment:
Hmm, am I missing something? If we fail you just log the message and if we don't fail you ... stop the ticker?